src/librustdoc/doctest.rs

```diff
@@ -348,13 +351,49 @@ fn run_test(
         .unwrap_or_else(|| rustc_interface::util::rustc_path().expect("found rustc"));
     let mut compiler = wrapped_rustc_command(&rustdoc_options.test_builder_wrappers, rustc_binary);
     compiler.arg("--crate-type").arg("bin");
-    for cfg in &rustdoc_options.cfgs {
-        compiler.arg("--cfg").arg(&cfg);
+
+    let mut needs_unstable_options = false;
+    let mut written_cfg_args = false;
+    #[allow(unused_variables)]
+    let tmp_dir;
+
+    // If there are too many `cfg` arguments, instead of risking reaching `Command`'s limit on
+    // the number of arguments, we put them into a file which we then pass as `@` argument.
+    if rustdoc_options.cfgs.len() + rustdoc_options.check_cfgs.len() > 1_000
```

Comment:
Thing can become unreadable and have a perf impact way lower, I suggest switching at 50. It should be enough to handle most crates without the argfile while switch sooner rather than latter for bigger crate.
```suggestion
    if rustdoc_options.cfgs.len() + rustdoc_options.check_cfgs.len() > 50
```

Reply:
Yeah, some operating systems may have low limits on the number of arguments and/or total length, so it is best to be conservative.

Reply:
At this point I think I'll simply put all arguments into a file directly and pass it to all doctests. :)